store/src/java/com/zimbra/cs/account/EmailChannel.java

```diff
@@ -184,10 +185,10 @@ public static void sendAndStoreResetPasswordURL(ZimbraSoapContext zsc, Account a
         String userDisplayName = account.getDisplayName() != null ? String.join("", " ", account.getDisplayName()) : "";
         String subject = L10nUtil.getMessage(MsgKey.sendPasswordResetEmailSubject, locale, userDisplayName);
         String charset = account.getAttr(Provisioning.A_zimbraPrefMailDefaultCharset, MimeConstants.P_CHARSET_UTF8);
-        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("EEE, d MMM yyyy HH:mm:ss z").withZone(ZoneId.of("GMT"));
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("EEE, d MMM yyyy HH:mm:ss z").withZone(ZoneId.of(Util.getAccountTimeZone(account).getID()));
```

Comment:
Save the result of the `withZone` argument and use below with `atZone` instead of recalculating.